packages/sources/elven/src/endpoint/reserves.ts

```diff
@@ -0,0 +1,138 @@
+  return (
+    Array.isArray(values) &&
+    values.every(
+      (value) =>
+        value != null &&
+        typeof value === 'object' &&
+        'amount' in value &&
+        'fiat_amount' in value &&
+        'address' in value &&
+        typeof value.amount === 'number' &&
+        typeof value.fiat_amount === 'number' &&
+        typeof value.address === 'string',
+    )
+  )
+}
+
+const parseAssetReserves = (res: ProviderResponseBody['data']): Record<string, AssetReserves[]> => {
+  const assetReserves: Record<string, AssetReserves[]> = {}
+
+  Object.keys(res).forEach((key) => {
+    const value = res[key] as unknown
+    if (isAssetReservesList(value)) {
+      assetReserves[key] = value
+    }
+  })
+
+  return assetReserves
+}
+
+type HttpTransportTypes = EndpointTypes & {
+  Provider: {
+    RequestBody: undefined
+    ResponseBody: ProviderResponseBody
+  }
+}
+
+const httpTransport = new HttpTransport<HttpTransportTypes>({
+  prepareRequests: (params, config) => {
+    return {
+      params,
+      request: {
+        url: config.API_URL,
+        headers: {
+          'X-POR-ACCESS-KEY': config.API_KEY,
+          'X-POR-ACCESS-SECRET': config.API_SECRET,
+        },
+      },
+    }
+  },
+  parseResponse: (params, res) => {
+    return params.map((param) => {
+      if (res.data.status.toLowerCase() != 'success') {
+        return {
+          params: param,
+          response: {
+            statusCode: 502,
+            errorMessage: `Invalid status returned from provider: '${res.data.status}'.`,
+          },
+        }
+      }
+
+      const data = res.data?.data
+      if (!data || !data.total_value_in_usd) {
+        return {
+          params: param,
+          response: {
+            statusCode: 502,
+            errorMessage: `Invalid data returned from provider - does not contain total reserves value.`,
+          },
+        }
+      }
+
+      const assetReserves = parseAssetReserves(data)
+      const aggregate_fiat_value = Object.values(assetReserves)
+        .map((reserves) => reserves.map((reserve) => reserve.fiat_amount))
+        .flat()
+        .reduce((sum, current) => sum + current, 0)
+
+      if (Math.round(aggregate_fiat_value) != Math.round(data.total_value_in_usd)) {
+        return {
+          params: param,
+          response: {
+            statusCode: 502,
+            errorMessage: `Invalid total_value_in_usd returned from provider - does not match aggregated reserves value.`,
+          },
+        }
+      }
+
+      return {
+        params: param,
+        response: {
+          data: {
+            result: data.total_value_in_usd,
+          },
+          result: data.total_value_in_usd,
+          timestamps: {
+            providerIndicatedTimeUnixMs: new Date(data.timestamp * 1000).getTime(),
+          },
+        },
+      }
+    })
+  },
+})
+
+export const endpoint = new AdapterEndpoint({
+  name: 'reserves',
```

Comment:
My thinking was to keep the adapter & endpoint generic for Elven, so that by just updating the `API_ENDPOINT` the Elven EA could be used for a separate project.

I don't think there is anything specific to `hope` that requires the endpoint to be specific for that case, and so would prefer to keep a generic name. 

Interested in your thoughts though, as I might not be clear on how NOPs will run this EA if it was generic to multiple projects (e.g. running 1 EA with multiple endpoints vs running 2 instances of the same EA with different config).

Reply:
I think the preferred way to run it would be as a single instance with multiple endpoints. I believe we made `armanino` the same way. The endpoint will be specific to Hope once you make the change above to move the `/hope-money` path to the endpoint. There's also a chance another project may not need the same value and could have a different response all together so we maybe can't rely on just changing the `API_ENDPOINT`.

Reply:
That makes sense - I’ll tweak so that the endpoint contains all the hope’-specific parts